src/app/src/components/StackedBarChart.js

```diff
@@ -2,45 +2,55 @@ import React from 'react';
 import { useSelector } from 'react-redux';
 import { Box } from '@chakra-ui/react';
 import { ResponsiveBarCanvas } from '@nivo/bar';
+import sortBy from 'lodash/sortBy';
+import flatMap from 'lodash/flatMap';
 
 import DownloadMenu from './DownloadMenu';
 import useRefs from '../hooks/useRefs';
 import { formatStackedCSV } from '../utils/csv';
 
+const formatResponsesForChart = ({ question, responses, sortedYears }) => {
+    let data = sortedYears
+        .map(year => {
+            const validResponses = responses
+                .filter(r => r.year === year)
+                .filter(r => !r.dataUnavailable);
+            return validResponses.length
+                ? validResponses.reduce(
+                      (acc, curr) => {
+                          acc[0][curr.cat] = Math.round(curr.combined, 2);
+                          acc[1][curr.cat] = Math.round(curr.male, 2);
+                          acc[2][curr.cat] = Math.round(curr.female, 2);
+                          return acc;
+                      },
+                      [
+                          { index: `Total ${year}` },
+                          { index: `Men ${year}` },
+                          { index: `Women ${year}` },
+                      ]
+                  )
+                : validResponses;
```

Comment:
Presumably this is the case when `validResponses.length` is `0`? Consider returning an explicit empty array to make that clearer.

```suggestion
                : [];
```